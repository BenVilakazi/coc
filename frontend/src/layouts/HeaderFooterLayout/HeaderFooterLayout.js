import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Footer from './Footer/Footer';
import config from '../../config';

const propTypes = {
  isWelcoming: PropTypes.bool,
  children: PropTypes.node,
};

const defaultProps = {
  isWelcoming: false,
  children: '',
};

const { largeMobileWidth } = config.breakpoint.playerBreakpoints;

const Header = styled.header`
  height: min(20vh, 400px);
  width: 100%;
  background-color: var(--secondary-background-color);
  color: var(--secondary-text-color);
  display: flex;
  flex-direction: column;
  justify-content: flex-end;

  .welcome-to {
    display: none;
  }

  .CoC {
    font-size: clamp(48px, 4.5vw, 110px);
    font-weight: 700;
    line-height: 100%;
    padding-right: 5px;
    margin-bottom: -7px;
  }

  @media (min-width: 491px) {
    .welcome-to {
      display: flex;
      font-size: clamp(18px, 1.7vw, 42px);
      margin-bottom: max(-0.7vw, -10px);
      font-weight: 500;
      line-height: 100%;
    }
  }

  @media (min-width: ${largeMobileWidth}) {
    .CoC {
      margin-bottom: -8px;
      margin-bottom: max(-0.8vw, -20px);
    }
  }

  @media (orientation: landscape) and (max-height: 500px) {
    height: 80px;
  }
`;

function HeaderFooterLayout({ isWelcoming, children }) {
  return (
    <>
      <Header data-testid="header">
        {isWelcoming && (
          <p data-testid="welcome-text" className="welcome-to">
            WELCOME TO
          </p>
        )}
        <h1 className="CoC">CARDS OF CAROUSAL</h1>
      </Header>
      {children}
      <Footer isWelcoming={isWelcoming} />
    </>
  );
}

HeaderFooterLayout.propTypes = propTypes;
HeaderFooterLayout.defaultProps = defaultProps;

export default HeaderFooterLayout;
