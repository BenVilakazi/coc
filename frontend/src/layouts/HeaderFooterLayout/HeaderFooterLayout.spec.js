import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import HeaderFooterLayout from './HeaderFooterLayout';

describe('HeaderFooterLayout', () => {
  describe('render', () => {
    it('renders correctly', () => {
      const tree = renderer
        .create(<HeaderFooterLayout>Test</HeaderFooterLayout>)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders correctly with the isWelcoming prop', () => {
      const tree = renderer
        .create(<HeaderFooterLayout isWelcoming>Test</HeaderFooterLayout>)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders children', () => {
      render(<HeaderFooterLayout>Test</HeaderFooterLayout>);

      expect(screen.getByText('Test')).toBeInTheDocument();

      expect(screen.getByTestId('header')).toBeInTheDocument();
      expect(screen.getByTestId('footer')).toBeInTheDocument();
    });

    it('renders when no children are given', () => {
      render(<HeaderFooterLayout />);

      expect(screen.getByTestId('header')).toBeInTheDocument();
      expect(screen.getByTestId('footer')).toBeInTheDocument();
    });

    it('displays the welcome text when the isWelcoming prop is given', () => {
      render(<HeaderFooterLayout isWelcoming>Test</HeaderFooterLayout>);

      expect(screen.getByTestId('welcome-text')).toBeInTheDocument();
    });

    it('does not display the welcome text when the isWelcoming prop is not given', () => {
      render(<HeaderFooterLayout>Test</HeaderFooterLayout>);

      expect(screen.queryByTestId('welcome-text')).not.toBeInTheDocument();
    });
  });
});
