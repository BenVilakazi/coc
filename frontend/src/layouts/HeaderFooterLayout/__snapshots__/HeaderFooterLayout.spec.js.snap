// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderFooterLayout render renders correctly 1`] = `
Array [
  .c0 {
  height: min(20vh,400px);
  width: 100%;
  background-color: var(--secondary-background-color);
  color: var(--secondary-text-color);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c0 .welcome-to {
  display: none;
}

.c0 .CoC {
  font-size: clamp(48px,4.5vw,110px);
  font-weight: 700;
  line-height: 100%;
  padding-right: 5px;
  margin-bottom: -7px;
}

@media (orientation:portrait) {

}

@media (min-width:) {

}

@media (min-width:) {

}

@media (min-width:) {

}

@media (min-width:491px) {
  .c0 .welcome-to {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    font-size: clamp(18px,1.7vw,42px);
    margin-bottom: max(-0.7vw,-10px);
    font-weight: 500;
    line-height: 100%;
  }
}

@media (min-width:600px) {
  .c0 .CoC {
    margin-bottom: -8px;
    margin-bottom: max(-0.8vw,-20px);
  }
}

@media (orientation:landscape) and (max-height:500px) {
  .c0 {
    height: 80px;
  }
}

<header
    className="c0"
    data-testid="header"
  >
    <h1
      className="CoC"
    >
      CARDS OF CAROUSAL
    </h1>
  </header>,
  "Test",
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  text-align: center;
  height: 50px;
  font-size: 1rem;
  font-weight: 400;
  padding-right: 10px;
  padding-bottom: 4px;
  background-color: var(--secondary-background-color);
  color: var(--secondary-text-color);
}

.c0 p {
  text-align: right;
  position: relative;
  font-size: 13px;
}

.c0 a {
  color: var(--secondary-text-color);
  -webkit-text-decoration: none;
  text-decoration: none;
}

@media (orientation:portrait) {
  .c0 {
    height: 120px;
    background-color: transparent;
    color: var(--accent-text-color);
  }

  .c0 p {
    text-align: center;
    padding-bottom: 32px;
  }

  .c0 a {
    color: var(--accent-text-color);
    display: block;
  }
}

@media (min-width:) {
  .c0 {
    height: 90px;
  }

  .c0 p,
  .c0 a {
    font-size: 20px;
  }
}

@media (min-width:) {
  .c0 {
    height: 110px;
  }

  .c0 p,
  .c0 a {
    font-size: 30px;
  }
}

@media (min-width:) {
  .c0 {
    height: 122px;
    padding-right: 16px;
    padding-bottom: 2px;
  }

  .c0 p,
  .c0 a {
    font-size: 36px;
  }
}

@media (min-width:491px) {

}

@media (min-width:600px) {

}

@media (orientation:landscape) and (max-height:500px) {

}

<footer
    className="c0"
    data-testid="footer"
  >
    <p>
      Card content thanks to:
       
      <a
        href="https://cardsagainsthumanity.com"
      >
        https://cardsagainsthumanity.com
      </a>
    </p>
    <p>
      Made by Odin students, with love:
       
      <a
        href="https://www.theodinproject.com"
      >
        https://www.theodinproject.com
      </a>
    </p>
  </footer>,
]
`;

exports[`HeaderFooterLayout render renders correctly with the isWelcoming prop 1`] = `
Array [
  .c0 {
  height: min(20vh,400px);
  width: 100%;
  background-color: var(--secondary-background-color);
  color: var(--secondary-text-color);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c0 .welcome-to {
  display: none;
}

.c0 .CoC {
  font-size: clamp(48px,4.5vw,110px);
  font-weight: 700;
  line-height: 100%;
  padding-right: 5px;
  margin-bottom: -7px;
}

@media (orientation:portrait) {

}

@media (min-width:) {

}

@media (min-width:) {

}

@media (min-width:) {

}

@media (min-width:491px) {
  .c0 .welcome-to {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    font-size: clamp(18px,1.7vw,42px);
    margin-bottom: max(-0.7vw,-10px);
    font-weight: 500;
    line-height: 100%;
  }
}

@media (min-width:600px) {
  .c0 .CoC {
    margin-bottom: -8px;
    margin-bottom: max(-0.8vw,-20px);
  }
}

@media (orientation:landscape) and (max-height:500px) {
  .c0 {
    height: 80px;
  }
}

<header
    className="c0"
    data-testid="header"
  >
    <p
      className="welcome-to"
      data-testid="welcome-text"
    >
      WELCOME TO
    </p>
    <h1
      className="CoC"
    >
      CARDS OF CAROUSAL
    </h1>
  </header>,
  "Test",
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  text-align: center;
  height: 50px;
  font-size: 1rem;
  font-weight: 400;
  padding-right: 10px;
  padding-bottom: 4px;
  background-color: var(--secondary-background-color);
  color: var(--secondary-text-color);
}

.c0 p {
  text-align: right;
  position: relative;
  font-size: 13px;
}

.c0 a {
  color: var(--secondary-text-color);
  -webkit-text-decoration: none;
  text-decoration: none;
}

@media (orientation:portrait) {
  .c0 {
    height: 120px;
    background-color: transparent;
    color: var(--accent-text-color);
  }

  .c0 p {
    text-align: center;
    padding-bottom: 32px;
  }

  .c0 a {
    color: var(--accent-text-color);
    display: block;
  }
}

@media (min-width:) {
  .c0 {
    height: 90px;
  }

  .c0 p,
  .c0 a {
    font-size: 20px;
  }
}

@media (min-width:) {
  .c0 {
    height: 110px;
  }

  .c0 p,
  .c0 a {
    font-size: 30px;
  }
}

@media (min-width:) {
  .c0 {
    height: 122px;
    padding-right: 16px;
    padding-bottom: 2px;
  }

  .c0 p,
  .c0 a {
    font-size: 36px;
  }
}

@media (min-width:491px) {

}

@media (min-width:600px) {

}

@media (orientation:landscape) and (max-height:500px) {

}

<footer
    className="c0"
    data-testid="footer"
  >
    <p>
      Card content thanks to:
       
      <a
        href="https://cardsagainsthumanity.com"
      >
        https://cardsagainsthumanity.com
      </a>
    </p>
    <p>
      Made by Odin students, with love:
       
      <a
        href="https://www.theodinproject.com"
      >
        https://www.theodinproject.com
      </a>
    </p>
  </footer>,
]
`;
